app/src/main/java/net/gaast/giggity/Schedule.java

```diff
@@ -421,6 +424,155 @@ private void loadIcal(BufferedReader in) {
 		}
 	}
 
+	/*Reading the JSON open event format data fetched from the API end point*/
+	private void loadJson(BufferedReader in) {
+
+		StringBuffer buffer = new StringBuffer();
+		SimpleDateFormat df = new SimpleDateFormat();
+		HashMap<String, Schedule.Line> tentMap = new HashMap<String, Schedule.Line>();
+		Scanner s = new Scanner(in);
+
+		while (s.hasNext()) {
+			buffer.append(s.nextLine());
+
+		}
+
+		String output = buffer.toString();
+		Schedule.Line line = null;
+
+		try {
+
+			JSONObject conference = new JSONObject(output);
```

Comment:
Slurping the whole thing into a StringBuffer, then string, then this parsing makes me sad. But it looks like the methods on https://stackoverflow.com/questions/22461663/convert-inputstream-to-jsonobject and https://stackoverflow.com/questions/31217653/how-to-read-and-write-json-using-json-simple-library work with different JSON parsers, not in Android's core API so they would pull in extra compiled-in dependencies?

I may do some experimenting once this is merged. This is not a big deal but it just looks less clean to me.